disnake/ui/action_row.py

```diff
@@ -24,86 +24,170 @@
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, List, Optional, Sequence, TypeVar, Union
-
-from ..components import ActionRow as ActionRowComponent, NestedComponent, SelectOption
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    ClassVar,
+    Generator,
+    Generic,
+    List,
+    Literal,
+    Optional,
+    Sequence,
+    Tuple,
+    TypeVar,
+    Union,
+    overload,
+)
+
+from ..components import (
+    ActionRow as ActionRowComponent,
+    Button as ButtonComponent,
+    NestedComponent,
+    SelectMenu as SelectComponent,
+    SelectOption,
+)
 from ..enums import ButtonStyle, ComponentType, TextInputStyle
 from ..utils import MISSING
 from .button import Button
 from .item import WrappedComponent
 from .select import Select
 from .text_input import TextInput
 
-__all__ = ("ActionRow",)
-
 if TYPE_CHECKING:
     from ..emoji import Emoji
+    from ..message import Message
     from ..partial_emoji import PartialEmoji
     from ..types.components import ActionRow as ActionRowPayload
 
-    ActionRowT = TypeVar("ActionRowT", bound="ActionRow")
-    Components = Union[
-        ActionRowT,
-        WrappedComponent,
-        Sequence[Union[ActionRowT, WrappedComponent, Sequence[WrappedComponent]]],
-    ]
+__all__ = (
+    "ActionRow",
+    "Components",
+    "MessageUIComponent",
+    "ModalUIComponent",
+    "MessageActionRow",
+    "ModalActionRow",
+)
+
 
+MessageUIComponent = Union[Button[Any], Select[Any]]
+ModalUIComponent = Union[TextInput, Select[Any]]
```

Comment:
I think that selects should temporarily be left out until full support is added to modal interactions.

Reply:
Alright, that's fair. I'll revert that for now, then. Re-adding support is a quick 2-line change anyways.

Reply:
All the changes we need to make are left as comments so we can easily re-enable them, minus the docstring change.